dask/array/_array_expr/_expr.py

```diff
@@ -117,7 +140,10 @@ def rechunk(
 
         from dask.array._array_expr._rechunk import Rechunk
 
-        return Rechunk(self, chunks, threshold, block_size_limit, balance, method)
+        result = Rechunk(self, chunks, threshold, block_size_limit, balance, method)
+        # Ensure that chunks are compatible
+        result.chunks
```

Comment:
This is a bit odd. Is this intended to stay here? If so, I wonder if this raises a comprehensive error message

Reply:
rechunk checks if the new chunks are matching the shape and raises an informative error if not. Checking the chunks here will bobble the error up immediately rather than when we materialize later